core/src/consensus/sortition/lot.rs

```diff
@@ -0,0 +1,31 @@
+// Copyright 2019 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use rug::{integer::Order, Integer, Rational};
+
+use super::binom_cdf::binomial_cdf_walk;
+
+pub fn lot(voting_power: u64, total_power: u64, expectation: f64, vrf_output: &[u8]) -> u64 {
```

Comment:
What do you think about adding a simple description of the result?

Reply:
I added an explanation document for this function.